### PR TITLE
[RFR] Sort by date & supporters amount

### PR DESCRIPTION
**Pull request by Tom-Bonnike**

### CODE REVIEW

File: src/actions/PetitionsActions.js [hunk 1]
```diff
@@ -19,14 +19,15 @@ export function fetchPetitions ({ location, params }) {
     page: parseInt(params && params.page || query.page || 1),
     city: params && params.city || query.city || '',
     cityName: params && params.cityName || query.cityName || '',
-    limit: parseInt(query.limit || 12)
+    limit: parseInt(query.limit || 12),
+    sort: params && params.sort || query.sort || ''
```
Comment (Tom-Bonnike) [inline src/actions/PetitionsActions.js]: oh, I guess I don’t need the tests for params. Sorting will only be done with query parameters

Comment (Tom-Bonnike) [inline src/actions/PetitionsActions.js]: oh, I guess I don’t need the tests for params. Sorting will only be done with query parameters

File: src/containers/Petitions.js [hunk 2]
```diff
@@ -22,43 +19,21 @@ const PetitionsContainer = withRouter(React.createClass({
   },
 
   componentWillReceiveProps (nextProps) {
-    if (this.props.location.pathname !== nextProps.location.pathname) {
+    if (this.props.location.pathname !== nextProps.location.pathname ||
+        this.props.location.search !== nextProps.location.search) {
```
Comment (Tom-Bonnike) [inline src/containers/Petitions.js]: @mattberridge 
Any better idea?

File: src/helpers/getPetitionsRequestParams.js [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
 
-export default ({ limit, page, city }) => {
+export default ({ limit, page, city, sort }) => {
   const saneLimit = sanitizeParamLimit(limit);
 
+  if (sort === 'supporters') {
+    sort = '-supporters.amount';
+  } else {
+    sort = '-created';
+  }
```
Comment (Tom-Bonnike) [inline src/helpers/getPetitionsRequestParams.js]: I guess for now we only need those two conditions, as the default is the same as “Most recent”

File: src/actions/PetitionsActions.js (hunk 1 above)
Comment (mattberridge) [inline src/actions/PetitionsActions.js]: just `query` :)

File: src/components/Petitions/index.js [hunk 4]
```diff
@@ -8,35 +8,29 @@ import PetitionsFilters from 'components/PetitionsFilters';
 import NoResults from 'components/NoResults';
 import Pagination from 'containers/Pagination';
 
-const Petitions = ({
-  petitions,
-  isLoading,
-  title,
-  autocompleteProps
-}) => (
+const Petitions = (props) => (
```
Comment (mattberridge) [inline src/components/Petitions/index.js]: why the style change?

Comment (Tom-Bonnike) [inline src/components/Petitions/index.js]: … I‘ll revert that. There was probably a reason, when I did it :p

Comment (Tom-Bonnike) [inline src/components/Petitions/index.js]: Ah… I remember now :) I need to pass all the props to the children PetitionsFilters component now, so I didn’t destructure the props object.

**Pull request merged**